components/registry-facade/cmd/run.go

```diff
@@ -43,6 +45,34 @@ var runCmd = &cobra.Command{
 			log.WithError(err).WithField("filename", configPath).Fatal("cannot load config")
 		}
 
+		promreg := prometheus.NewRegistry()
+		gpreg := prometheus.WrapRegistererWithPrefix("gitpod_registry_facade_", promreg)
+		rtt, err := registry.NewMeasuringRegistryRoundTripper(newDefaultTransport(), prometheus.WrapRegistererWithPrefix("downstream_", gpreg))
+		if err != nil {
+			log.WithError(err).Fatal("cannot registry metrics")
+		}
+		if cfg.PrometheusAddr != "" {
+			promreg.MustRegister(
+				collectors.NewGoCollector(),
+				collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+				common_grpc.ClientMetrics(),
+			)
+
+			handler := http.NewServeMux()
+			handler.Handle("/metrics", promhttp.HandlerFor(promreg, promhttp.HandlerOpts{}))
```

Comment:
I see that you create a new registry called `gpreg` to add the "gitpod_registry_facade" prefix, but then you use the previous registry, `promreg`, to serve the metrics endpoint.

Is that intended?

Reply:
I know that you just moved code to another location, it's just that it looks awkward 😅

Reply:
`gpreg` is used in two places and just wraps the original registerer. I don't think there's much to be changed here.